sources/packages/api/src/services/disbursement-receipt/disbursement-receipt.service.ts

```diff
@@ -123,4 +123,49 @@ export class DisbursementReceiptService extends RecordDataModelService<Disbursem
       .getRawOne();
     return batchRunDate?.max ?? new Date();
   }
+
+  /**
+   * Get the disbursement receipt details for
+   * given assessment.
+   * @param assessmentId assessment to which disbursement
+   * receipt belongs to.
+   * @param studentId student to whom the disbursement
+   * receipt belongs to.
+   * @returns disbursement receipt details.
+   */
+  async getDisbursementReceiptByAssessment(
+    assessmentId: number,
+    studentId?: number,
+  ): Promise<DisbursementReceipt[]> {
+    const findCriteria: FindManyOptions<DisbursementReceipt> = {
+      select: {
+        id: true,
+        disbursementSchedule: { id: true },
+        disbursementReceiptValues: { grantType: true, grantAmount: true },
+      },
+      relations: {
+        disbursementReceiptValues: true,
+        disbursementSchedule: true,
+      },
+    };
+    if (studentId) {
```

Comment:
The student id can be "appended" to the object in the same way that we would be doing with Typeorm query builder.
There is no need to repeat the code.

Reply:
I tried doing that. but couldn't achieve it in a typed fashion. May be we can sync up and see what could be done.

Reply:
Can you please give it a try in the below way?
![image](https://user-images.githubusercontent.com/61259237/191595102-272ac594-a238-4c69-83d2-eb1442cb2ba4.png)